src/Core/Managed/Shared/ActivityProxy.cs

```diff
@@ -0,0 +1,21 @@
+﻿namespace Microsoft.ApplicationInsights
+{
+    using System;
+    using System.Diagnostics;
+
+    internal static class ActivityProxy
+    {
+        public static bool TryRun(Func<bool> method)
+        {
+            Debug.Assert(method != null, "Method must not be null");
+            try
+            {
+                return method.Invoke();
+            }
+            catch (System.IO.FileNotFoundException)
```

Comment:
This needs an explanatory comment. What is ```TryRun``` used for and why is FileNotFoundException handled specially?